Tests/ConfigurationServiceSpec.swift

```diff
@@ -124,59 +124,97 @@ class ConfigurationServiceSpec: QuickSpec {
                         }
                     }
                 }
-            }
 
-            context("when request fails") {
-                beforeEach {
-                    httpSession.responseError = NSError(domain: "config.error.test", code: 1, userInfo: nil)
-                }
+                context("and status code is 4xx") {
 
-                it("will return ConfigurationServiceError containing .requestError") {
-                    waitUntil { done in
-                        requestQueue.async {
-                            let result = service.getConfigData()
-                            let error = result.getError()
-                            expect(error).toNot(beNil())
+                    it("will return ConfigurationServiceError containing .tooManyRequestsError for code 429") {
+                        httpSession.httpResponse = HTTPURLResponse(url: configURL,
+                                                                   statusCode: 429,
+                                                                   httpVersion: nil,
+                                                                   headerFields: nil)
```

Comment:
To make it more concise you could consider having a `ConfigURLResponse(statusCode: 429)` helper.